File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
//loop through the elements
// start adding if it is a 1, otherwise initialize it to zero

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0;
        int temp{};
        for(int n:nums){
            if (n==1) count +=1;
            else{
                temp = (count>temp)?count:temp;
                count = 0;
            }
        }
        temp = (count>temp)?count:temp;
        return temp;
}
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        length = 0
        maxLength = 0
        i = 0
        for j in range(len(nums)):
            if nums[j]:
                length += 1
                maxLength = max(maxLength,length)
            else:
                length = 0
        return maxLength